src/Robo.php

```diff
@@ -88,14 +90,36 @@ public static function hasContainer()
     }
 
     /**
+     * Create a config object and load it from the provided paths.
+     */
+    public static function createConfiguration($paths)
+    {
+        $config = new Config();
+        static::loadConfiguration($config, $paths);
+        return $config;
+    }
+
+    /**
+     * Use a simple config loader to load configuration values from specified paths
+     */
+    public static function loadConfiguration($config, $paths)
+    {
+        $loader = new YamlConfigLoader();
+        foreach ($paths as $path) {
+            $loader->load($path);
```

Comment:
I see. I thought that was happening in extend. As you mentioned, I should add more tests that include overridden configuration to ensure this works correctly.

My thought now is that there should be three classes, each with distinct responsibilities:

- Config loader: Responsible only for loading, from yaml files or whatever. Does any necessary transformations (e.g. dot-notation key expansion) to produce a nested array.
- Config merger: Takes any number of nested arrays. Merges them together and then does property expansion.
- Config class: Initialized with a nested array from a loader / merger. Provides simple get/set access for use at runtime.

If you need to load configuration in multiple stages, then you would export a nested array from the config object, use a config merger to combine the nested arrays, and then re-insert the result into the config object. This reduces coupling between the config classes, allowing folks to drop in Symfony/Config for their config loader if they need those features.